api/app/controllers/spree/api/base_controller.rb

```diff
@@ -45,33 +48,37 @@ def map_nested_attributes_keys(klass, attributes)
         end.with_indifferent_access
       end
 
+      # users should be able to set price when importing orders via api
+      def permitted_line_item_attributes
+        if current_api_user.has_spree_role?("admin")
+          super + [:price, :variant_id, :sku]
+        else
+          super
+        end
+      end
+
       private
 
       def set_content_type
         content_type = case params[:format]
         when "json"
-          "application/json"
+          "application/json; charset=utf-8"
         when "xml"
-          "text/xml"
+          "text/xml; charset=utf-8"
         end
         headers["Content-Type"] = content_type
       end
 
-      def check_for_user_or_api_key
-        # User is already authenticated with Spree, make request this way instead.
-        return true if @current_api_user = try_spree_current_user || !Spree::Api::Config[:requires_authentication]
-
-        if api_key.blank?
-          render "spree/api/errors/must_specify_api_key", :status => 401 and return
-        end
+      def load_user
+        @current_api_user = (try_spree_current_user || Spree.user_class.find_by(spree_api_key: api_key.to_s))
       end
 
       def authenticate_user
         unless @current_api_user
-          if requires_authentication? || api_key.present?
-            unless @current_api_user = Spree.user_class.find_by_spree_api_key(api_key.to_s)
-              render "spree/api/errors/invalid_api_key", :status => 401 and return
-            end
+          if requires_authentication? && api_key.blank? && order_token.blank?
+            render "spree/api/errors/must_specify_api_key", :status => 401 and return
```

Comment:
Line is too long. [85/80]<br>Use `&&` instead of `and`.<br>Use the new Ruby 1.9 hash syntax.